### PR TITLE
Script to get build information

### PR DESCRIPTION
**Pull request by MCardenasBlanco**
With this script, you can get build information using groovy.
@reviewbybees

### CODE REVIEW

**Comment by carlosrodlop** [comment]
🐝

File: get-build-information.groovy [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+*/
+Jenkins.instance.getAllItems(Job).each{
+
+  def jobBuilds=it.getBuilds()
+
+	//for each of such jobs we can get all the builds (or you can limit the number at your convenience)
+    jobBuilds.each { build ->
+      def runningSince = groovy.time.TimeCategory.minus( new Date(), build.getTime() )
+      def currentStatus = build.getBuildStatusSummary().message
```
Comment (jglick) [inline get-build-information.groovy]: `.buildStatusSummary`, idiomatically

File: get-build-information.groovy [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      def cause = build.getCauses()[0] //we keep the first cause
```
Comment (jglick) [inline get-build-information.groovy]: `.causes`

Comment (jglick) [inline get-build-information.groovy]: Uh, what if `causes` is empty? You will get an `ArrayIndexOutOfBoundsException`.

File: get-build-information.groovy [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      //This is a simple case where we want to get information on the cause if the build was 
+      //triggered by an user
+      def user = cause instanceof Cause.UserIdCause? cause.getUserId():""
+      //This is an easy way to show the information on screen but can be changed at convenience
+      println "Build: ${build} | Since: ${runningSince} | Status: ${currentStatus} | Cause: ${cause} | User: ${user}" 
+     
+      // You can get all the information available for build parameters.
+      def parameters = build.actions.find{ it instanceof ParametersAction }?.parameters
```
Comment (jglick) [inline get-build-information.groovy]: `build.getAction(ParametersAction)`

File: get-build-information.groovy [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+      parameters.each {
```
Comment (jglick) [inline get-build-information.groovy]: NPE if not parameterized.

Comment (MCardenasBlanco) [inline get-build-information.groovy]: Used in a non-parametrized job, and no NPE appeared. I think that in Groovy the each over a null object returns an empty list.

File: get-build-information.groovy (hunk 1 above)
Comment (MCardenasBlanco) [inline get-build-information.groovy]: Corrected.

File: get-build-information.groovy (hunk 3 above)
Comment (MCardenasBlanco) [inline get-build-information.groovy]: Corrected

File: get-build-information.groovy (hunk 2 above)
Comment (MCardenasBlanco) [inline get-build-information.groovy]: It seems that Groovy treats the collections in a different way. I tested with a null cause and no exception was thrown.

**Comment by MCardenasBlanco** [comment]
Thank you for your review @jglick, I think that I have addressed your comments

**Pull request merged**